src/backend/core/models.py

```diff
@@ -1345,3 +1346,195 @@ def get_dns_record_value(self) -> str:
+        _("is active"),
+        default=True,
+        help_text=_("Whether this template is available for use"),
+    )
+
+    maildomains = models.ManyToManyField(
+        "MailDomain",
+        through="EmailTemplateMailDomain",
+        related_name="email_templates",
+        help_text=_("Mail domains that can use this template"),
+    )
+
+    mailboxes = models.ManyToManyField(
+        "Mailbox",
+        through="EmailTemplateMailbox",
+        related_name="email_templates",
+        help_text=_("Mailboxes that can use this template"),
+    )
+
+    class Meta:
+        db_table = "messages_emailtemplate"
+        verbose_name = _("email template")
+        verbose_name_plural = _("email templates")
+        ordering = ["name"]
+
+    def __str__(self):
+        return f"{self.name} ({self.get_type_display()})"
+
+    def get_abilities(self, user):
+        """Get abilities for this template based on user permissions."""
+        abilities = []
+
+        # Check if user has access to any of the mailboxes
+        if user.is_superuser or self.mailboxes.filter(accesses__user=user).exists():
+            abilities.extend(
+                [
+                    UserAbilityChoices.CAN_VIEW_DOMAIN_ADMIN,
+                    UserAbilityChoices.CAN_CREATE_MAILDOMAINS,
+                ]
+            )
+
+        return abilities
+
+    def render_template(self, context: Dict[str, Any]) -> Dict[str, str]:
+        """
+        Render the template with the given context.
+
+        Args:
+            context: Dictionary of variables to substitute in the template
+
+        Returns:
+            Dictionary with 'html_body' and 'text_body' keys containing rendered content
+        """
+        rendered_html_body = self.html_body
+        rendered_text_body = self.text_body
+
+        # Simple placeholder substitution
+        for key, value in context.items():
+            placeholder = f"{{{key}}}"
+            rendered_html_body = rendered_html_body.replace(placeholder, str(value))
+            rendered_text_body = rendered_text_body.replace(placeholder, str(value))
+
+        return {
+            "html_body": rendered_html_body,
+            "text_body": rendered_text_body,
+        }
+
+    def get_formatted_content(self) -> str:
+        """
+        Get the template content with proper formatting based on type.
+        Mainly used for signatures.
+
+        Returns:
+            Formatted template content
+        """
+        if self.type == TemplateTypeChoices.SIGNATURE:
+            # For signatures, prefer HTML body if available, otherwise text body
+            return self.html_body or self.text_body
+
+        # For non-signatures, return HTML body if available, otherwise text body
+        return self.html_body or self.text_body
+
+    def is_signature(self) -> bool:
+        """Check if this template is a signature."""
+        return self.type == TemplateTypeChoices.SIGNATURE
+
+    def is_template(self) -> bool:
```

Comment:
Template.is_template also feels sub-optimal naming

Reply:
Also it seems to never be called. And I'm not sure just for the admin we need a is_signature() function